<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Home · GenomicAnnotations.jl</title><link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css" rel="stylesheet" type="text/css"/><link href="https://fonts.googleapis.com/css?family=Lato|Roboto+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link href="assets/documenter.css" rel="stylesheet" type="text/css"/></head><body><nav class="toc"><h1>GenomicAnnotations.jl</h1><select id="version-selector" onChange="window.location.href=this.value" style="visibility: hidden"></select><form class="search" id="search-form" action="search/"><input id="search-query" name="q" type="text" placeholder="Search docs"/></form><ul><li class="current"><a class="toctext" href>Home</a><ul class="internal"></ul></li></ul></nav><article id="docs"><header><nav><ul><li><a href>Home</a></li></ul></nav><hr/><div id="topbar"><span>Home</span><a class="fa fa-bars" href="#"></a></div></header><p>Package doesn&#39;t contain Documenter docs.</p><p>Docs automatically generated by juliadocs.org</p><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.Locus" href="#GenomicAnnotations.Locus"><code>GenomicAnnotations.Locus</code></a> — <span class="docstring-category">Type</span>.</div><div><div><p>Struct for storing information on genomic locations. <code>strand</code> can be &#39;+&#39;, &#39;-&#39;, or &#39;.&#39; when the strand is irrelevant.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.Locus-Tuple{}" href="#GenomicAnnotations.Locus-Tuple{}"><code>GenomicAnnotations.Locus</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">Locus()
Locus(position::UnitRange{Int})
Locus(position::UnitRange{Int}, strand::Char)</code></pre></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="BioCore.sequence-Tuple{AbstractGene}" href="#BioCore.sequence-Tuple{AbstractGene}"><code>BioCore.sequence</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">sequence(gene::AbstractGene)</code></pre><p>Return genomic sequence for <code>gene</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.addgene!-Tuple{Chromosome,Any,Any}" href="#GenomicAnnotations.addgene!-Tuple{Chromosome,Any,Any}"><code>GenomicAnnotations.addgene!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">addgene!(chr::Chromosome, feature, locus; kw...)</code></pre><p>Add gene to <code>chr</code>. <code>locus</code> can be a <code>Locus</code>, a UnitRange, or a StepRange.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.iscomplement-Tuple{AbstractGene}" href="#GenomicAnnotations.iscomplement-Tuple{AbstractGene}"><code>GenomicAnnotations.iscomplement</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">iscomplement(gene::Abstract)</code></pre><p>Return <code>true</code> if <code>gene.locus.compliment == &#39;-&#39;</code>, otherwise return <code>false</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.pushproperty!-Union{Tuple{T}, Tuple{AbstractGene,Symbol,T}} where T" href="#GenomicAnnotations.pushproperty!-Union{Tuple{T}, Tuple{AbstractGene,Symbol,T}} where T"><code>GenomicAnnotations.pushproperty!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">pushproperty!(gene::AbstractGene, name::Symbol, x::T)</code></pre><p>Add a property to <code>gene</code>, similarly to <code>Base.setproperty!(::gene)</code>, but if the property is not missing in <code>gene</code>, it will be transformed to store a vector instead of overwriting existing data.</p><pre><code class="language-julia">julia&gt; eltype(chr.genedata[:EC_number])
Union{Missing,String}

julia&gt; chr.genes[1].EC_number = &quot;EC:1.2.3.4&quot;
&quot;EC:1.2.3.4&quot;

julia&gt; pushproperty!(chr.genes[1], :EC_number, &quot;EC:4.3.2.1&quot;); chr.genes[1].EC_number
2-element Array{String,1}:
 &quot;EC:1.2.3.4&quot;
 &quot;EC:4.3.2.1&quot;

julia&gt; eltype(chr.genedata[:EC_number])
Union{Missing, Array{String,1}}</code></pre></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.readgbk-Tuple{Any}" href="#GenomicAnnotations.readgbk-Tuple{Any}"><code>GenomicAnnotations.readgbk</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">readgbk(filename)</code></pre><p>Parse GenBank-formatted file <code>filename</code>, returning a <code>Chromosome</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Base.delete!-Tuple{AbstractArray{Gene,1}}" href="#Base.delete!-Tuple{AbstractArray{Gene,1}}"><code>Base.delete!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">delete!(genes::AbstractArray{Gene, 1})</code></pre><p>Delete all genes in <code>genes</code> from <code>genes[1].parent</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Base.delete!-Tuple{Gene}" href="#Base.delete!-Tuple{Gene}"><code>Base.delete!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">delete!(gene::Gene)</code></pre><p>Delete <code>gene</code> from <code>gene.parent</code>. Warning: does not work when broadcasted! Use <code>delete!(::AbstractVector{Gene}) instead.</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="Base.get-Tuple{Gene,Any,Any}" href="#Base.get-Tuple{Gene,Any,Any}"><code>Base.get</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">get(g::Gene, key, default)</code></pre><p>Retrieve <code>key</code> from <code>g</code>. If <code>key</code> is missing, return <code>default</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.filterseq-Tuple{Base.GenericIOBuffer{Array{UInt8,1}}}" href="#GenomicAnnotations.filterseq-Tuple{Base.GenericIOBuffer{Array{UInt8,1}}}"><code>GenomicAnnotations.filterseq</code></a> — <span class="docstring-category">Method</span>.</div><div><div><p>Parse footer (sequence) portion of a GenBank file, returning a <code>String</code>. When <code>BioSequences</code> is available for 0.7 this will be changed to a <code>DNASequence</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.parsechromosome-Tuple{Any}" href="#GenomicAnnotations.parsechromosome-Tuple{Any}"><code>GenomicAnnotations.parsechromosome</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">parsechromosome(lines)</code></pre><p>Parse and return one chromosome entry, and the line number that it ends at.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.parseposition-Tuple{String}" href="#GenomicAnnotations.parseposition-Tuple{String}"><code>GenomicAnnotations.parseposition</code></a> — <span class="docstring-category">Method</span>.</div><div><div><p>Parse lines encoding genomic position, returning the feature as a <code>String</code>, and an instance of <code>Locus</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GenomicAnnotations.vectorise-Union{Tuple{AbstractArray{Union{Missing, T},1}}, Tuple{T}} where T" href="#GenomicAnnotations.vectorise-Union{Tuple{AbstractArray{Union{Missing, T},1}}, Tuple{T}} where T"><code>GenomicAnnotations.vectorise</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">vectorise(A::AbstractArray{Union{Missing, T}, 1}) where T</code></pre><p>Convert an array of type Vector{Union{Missing, T}} to Vector{Union{Missing, Vector{T}}}.</p></div></div></section><footer><hr/></footer></article></body></html>
