<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Home · DPClustering.jl</title><link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css" rel="stylesheet" type="text/css"/><link href="https://fonts.googleapis.com/css?family=Lato|Roboto+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link href="assets/documenter.css" rel="stylesheet" type="text/css"/></head><body><nav class="toc"><h1>DPClustering.jl</h1><select id="version-selector" onChange="window.location.href=this.value" style="visibility: hidden"></select><form class="search" id="search-form" action="search/"><input id="search-query" name="q" type="text" placeholder="Search docs"/></form><ul><li class="current"><a class="toctext" href>Home</a><ul class="internal"></ul></li></ul></nav><article id="docs"><header><nav><ul><li><a href>Home</a></li></ul></nav><hr/><div id="topbar"><span>Home</span><a class="fa fa-bars" href="#"></a></div></header><p>Package doesn&#39;t contain Documenter docs.</p><p>Docs automatically generated by juliadocs.org</p><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="DPClustering.dpclustering-Tuple{Any,Any}" href="#DPClustering.dpclustering-Tuple{Any,Any}"><code>DPClustering.dpclustering</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">dpclustering(y::Array{Real, 1}, N::Array{Real, 1}; &lt;keyword arguments&gt;)</code></pre><p>Perform dirichlet clustering on the variant allele frequency distribution of cancer sequencing data and find the number of clusters that the data supports, y is a vector of the number of reads reporting each mutant, N is the total depth at each locus.</p><p>...</p><p><strong>Arguments</strong></p><ul><li><code>iterations = 1000</code>: number of iterations of the gibbs samples</li><li><code>C = 30</code>: Max number of clusters to consider</li><li><code>burninstart = round(Int64, iterations/2)</code>: Burn in of the gibbs samples</li><li><code>bw = 0.01</code>: Bandwidth of density estimation</li><li><code>maxxaxis = 0.7</code>:</li><li><code>cutoffweight = 0.05</code>: Minimum weight to be called a cluster</li><li><code>verbose = true</code>: Show progress of gibbs sampling with <code>ProgressMeter</code> package</li><li><code>A = 0.01</code>: Hyperparameter for α, see Nik-Zainal et al</li><li><code>B = 0.01</code>: Hyperparameter for α, see Nik-Zainal et al</li></ul><p>...</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="DPClustering.plotresults-Tuple{Any}" href="#DPClustering.plotresults-Tuple{Any}"><code>DPClustering.plotresults</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-none">plotresults(dp; &lt;keyword arguments&gt;)</code></pre><p>Plot results from DPClustering object. Will plot histogram of raw data with density estimates from Gibbs sampling. ...</p><p><strong>Arguments</strong></p><ul><li><code>save = false</code>: Set to <code>true</code> if you want the plot to be saved</li><li><code>dir = &quot;&quot;</code>: Directory where the plot will be saved to. Default is the current working directory.</li><li><code>plotname = &quot;DPclustering&quot;</code>: Name to call plot when saving.</li></ul><p>...</p></div></div></section><footer><hr/></footer></article></body></html>
