var documenterSearchIndex = {"docs": [

{
    "location": "#DPClustering.dpclustering-Tuple{Any,Any}",
    "page": "Home",
    "title": "DPClustering.dpclustering",
    "category": "method",
    "text": "dpclustering(y::Array{Real, 1}, N::Array{Real, 1}; <keyword arguments>)\n\nPerform dirichlet clustering on the variant allele frequency distribution of cancer sequencing data and find the number of clusters that the data supports, y is a vector of the number of reads reporting each mutant, N is the total depth at each locus.\n\n...\n\nArguments\n\niterations = 1000: number of iterations of the gibbs samples\nC = 30: Max number of clusters to consider\nburninstart = round(Int64, iterations/2): Burn in of the gibbs samples\nbw = 0.01: Bandwidth of density estimation\nmaxxaxis = 0.7:\ncutoffweight = 0.05: Minimum weight to be called a cluster\nverbose = true: Show progress of gibbs sampling with ProgressMeter package\nA = 0.01: Hyperparameter for α, see Nik-Zainal et al\nB = 0.01: Hyperparameter for α, see Nik-Zainal et al\n\n...\n\n\n\n\n\n"
},

{
    "location": "#DPClustering.plotresults-Tuple{Any}",
    "page": "Home",
    "title": "DPClustering.plotresults",
    "category": "method",
    "text": "plotresults(dp; <keyword arguments>)\n\nPlot results from DPClustering object. Will plot histogram of raw data with density estimates from Gibbs sampling. ...\n\nArguments\n\nsave = false: Set to true if you want the plot to be saved\ndir = \"\": Directory where the plot will be saved to. Default is the current working directory.\nplotname = \"DPclustering\": Name to call plot when saving.\n\n...\n\n\n\n\n\n"
},

{
    "location": "#",
    "page": "Home",
    "title": "Home",
    "category": "page",
    "text": "Package doesn\'t contain Documenter docs.Docs automatically generated by juliadocs.orgModules = [DPClustering]\nOrder = [:type, :function]"
},

]}
